disnake/app_commands.py

```diff
@@ -98,6 +98,11 @@ def _validate_name(name: str) -> None:
     # used for slash command names and option names
     # see https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-naming
 
+    if not isinstance(name, str):
+        raise TypeError(
+            f"slash command name and option names must be an instance of class 'str', received '{name.__class__}'"
```

Comment:
```suggestion
            f"Slash command name and option names must be an instance of class 'str', received '{name.__class__}'"
```
... to match the other exception's message.
Since it's not immediately evident from the stacktrace which name is invalid, it would be nice to add `name` itself to the message as well

Reply:
I don't think this is always called when `name` is the parameter name, tho, hence why it was slightly vague.